### ISSUE TITLE
Weapons/Arrows/Bolts embedded into monsters being destroyed by new looting system if the monster doesnt leave a corpse.

### ISSUE THREAD

**Issue by Cosmua** [CleverRaven/Cataclysm-DDA#53323]
### Describe the bug

While I like the looting change so far putting the zombies things onto the corpse for respawns and hopefully someone can get advanced inv and V to work with it I have found a possible unintended interaction/bug.

I first noticed it when I threw my combat knife at a gasoline zombie to finish it off. The knife became embeded and the zombie died shortly after to bleeding and exploded. But due to not leaving a corpse and loot/embeded items being in the corpse my embeded knife was gone too.

After this I wanted to test it so made a new world to debug my archery/marksmanship to 10 and fire some carbon fibre arrows into some different zombies, as I do enjoy playing with archery from time to time. Tough zombies worked pretty much as I expected, the arrows that became embeded were in the corpse after it died. The arrow that killed it was on the ground with the corpse, which is odd but okay sure.

However spawning in some boomers, gasoline zombies and such to try those I found all the embeded arrows get deleted with the corpse upon it exploding and only the arrow that landed the kill drops on the ground. While sure you could headcannon it that the explosion destroyed them all, it didnt use to work like that before and the embeded arrows (plus the other random loot drops) would be on the ground in older versions. So its a pretty nasty eater of valuable arrows, sure you could use lower quality ones I guess but idk.

TLDR: Due to embeded arrows/weapons being treated as loot thats now inside the corpse anything that doesnt leave the corpse behind means embeded weapons also get destroyed along with any loot those zombies would normally drop. I dont really know how github works so hopefully posting this here gets noticed. While idk i could see the explosion having a chance to destroy the arrows, its gonna suck to get a weapon embeded into anything that doesnt leave a corpse as your not getting that back.

### Steps to reproduce

Make a player with a bow and arrows (carbon fibre for easier results) and high skill.
Shoot at a Tough Zombie, or anything that leaves a corpse behind.

Repeat but shoot a Huge Boomer, or anything that explodes and/or doesnt leave a corpse on death.

### Expected behavior

Any arrows that become embeded should be "inside" the corpse with the monsters usual loot.

Note: All this also applies to if you get any weapon embedded in a monster, if it doesnt leave a corpse you cant get it back.

### Versions and configuration

Windows 10
Just the mods that are auto activated when making a new baseline world.
Version 9600194

**Comment by Alex-Folts** [comment]
It would be fun if embedded stuff after monster explode fly in all directions, i.e. apply throw action for each item and scatter them in random directions.

**Comment by Fris0uman** [comment]
Embeded arrows get aded to monster inventory there https://github.com/CleverRaven/Cataclysm-DDA/blob/c11797978e99a3d83d54825fe3bf5d01b2a50ed8/src/ballistics.cpp#L123

so the solution is probably to check if there's is a corpse to make around there https://github.com/CleverRaven/Cataclysm-DDA/blob/c11797978e99a3d83d54825fe3bf5d01b2a50ed8/src/monster.cpp#L2538 and drop the `inv` on the ground if there is none
